src/key/key.js

```diff
@@ -525,10 +528,17 @@ class Key {
     }
     // check for valid, unrevoked, unexpired self signature
     const { selfCertification } = await this.getPrimaryUser(date, userID, config);
-    // check for expiration time
+    // check for expiration time in binding signatures
     if (helper.isDataExpired(primaryKey, selfCertification, date)) {
       throw new Error('Primary key is expired');
     }
+    // check for expiration time in direct signatures
+    if (this.directSignatures.length > 0) {
+      const directSignature = await helper.getLatestValidSignature(this.directSignatures, primaryKey, enums.signature.key, { key: primaryKey }, date, config);
```

Comment:
This probably causes the key to be considered invalid if the direct signature is invalid, which might be an issue if someone adds a direct signature signed by a different key to an SKS server, for example.
So probably it would be good to add a try/catch around this line.
(Though I'm a bit confused, I thought this would cause [this test](https://github.com/openpgpjs/openpgpjs/blob/02a1ed2d783b89eee79e529f08d096f32debe911/test/general/key.js#L2751) to fail, but maybe I'm missing something?)